chia/server/node_discovery.py

```diff
@@ -311,7 +311,7 @@ async def _connect_to_peers(self, random) -> None:
                 elif len(groups) <= 5:
                     max_tries = 25
                 while not got_peer and not self.is_closed:
-                    sleep_interval = 1 + len(groups) * 0.5
+                    sleep_interval = len(groups) * 0.25
                     sleep_interval = min(sleep_interval, self.peer_connect_interval)
                     # Special case: try to find our first peer much quicker.
                     if len(groups) == 0:
```

Comment:
This can then be turned into `if not sleep_interval==0.0:` or just `if not sleep_interval:`.
(This is assuming `peer_connect_interval >= 0`, which should probably be assured somewhere)

Presumably `sleep_interval = max(sleep_interval, 1/10)` is what's seeked here.